src/main/java/picocli/CommandLine.java

```diff
@@ -365,7 +365,10 @@ public CommandLine addSubcommand(String name, Object command, String... aliases)
      * @since 0.9.7
      */
     public Map<String, CommandLine> getSubcommands() {
-        return new LinkedHashMap<String, CommandLine>(getCommandSpec().subcommands());
+        CaseAwareLinkedMap<String, CommandLine> subcommands = new CaseAwareLinkedMap<String, CommandLine>();
+        subcommands.setCaseInsensitive(isSubcommandsCaseInsensitive());
+        subcommands.putAll(getCommandSpec().subcommands());
```

Comment:
If we have a copy constructor that accepts a `CaseAwareLinkedMap` instead of a plain `Map`, then we can call `setCaseInsensitive` inside the constructor, based on whether the passed-in  `CaseAwareLinkedMap` parameter is case-sensitive.

That would reduce the above  code to simply:

```
return new CaseAwareLinkedMap<String, CommandLine>(getCommandSpec().commands);
```

Reply:
The issue here is that subcomands() actually returns an unmodifiable-map, which cannot be cast to CaseAwareLinkedMap without some nesty hacks. Not sure how to workaround that without breaking encapsulation.

Reply:
You can reference the `CommandSpec.commands` field directly, that has the right type.